tests/helpers/basetest.py

```diff
@@ -1,43 +1,32 @@
 from dataclasses import dataclass, field
-from typing import Optional, List
+from typing import Optional, List, Any
 import hashlib
 import base58
+from .deviceappproxy.deviceappproxy import DeviceAppProxy
+from ledgerblue.commException import CommException
+
 
 class NID:
     MAINNET = bytes.fromhex("00")
     TESTNET = bytes.fromhex("6f")
 
+
 class CONSENSUS_BRANCH_ID:
     OVERWINTER = bytes.fromhex("5ba81b19")
     SAPLING = bytes.fromhex("76b809bb")
     ZCLASSIC = bytes.fromhex("930b540d")
     ZCASH = bytes.fromhex("2BB40E60")
 
 
-@dataclass
-class LedgerjsApdu:
-    commands: List[str]
-    expected_resp: Optional[str] = field(default=None)
-    expected_sw: Optional[str] = field(default=None)
-    check_sig_format: Optional[bool] = field(default=None)
-
-@dataclass
-class TxData:
-    tx_to_sign: bytes
-    utxos: List[bytes]
-    output_paths: List[bytes]
-    change_path: bytes
-    expected_sig: List[bytes]
-
-
 @dataclass(init=False, repr=False)
```

Comment:
I initially planned more uses for the `PublicKey` class (which encapsulates the data returned by the app after a `GetPublicKey` command) like using its hash in the tx output script or using that key to verify the tx signature. But I finally didn't push further than parsing the pubkey data and displaying the various elements. `__repr__` was overloaded to display the significant pubkey fields from the raw data returned by the app. So yeah, it could be reworked as you propose.

Reply:
`BtcPublicKey` class rework marked as TODO for testsframeweor maintainers in test README